coremltools/converters/mil/frontend/_utils.py

```diff
@@ -39,6 +41,36 @@ def value_at(x: Var, idx: int, name=None, before_op=None):
     return mb.slice_by_index(**args)
 
 
+def _construct_gather_op(op_type: str, x: Var, indices: Var, axis: Var = None, **kwarg) -> Var:
```

Comment:
I prefer not to have "**kwarg" is possible.
For this case, we can just add `name` as an additional input?

Reply:
We can. We will need to wrap it ourselves in this util, though, because `mb` ops does not accept `name=None`

Concretely, we have 2 choices
1. Create something like `name_kwarg = {} if name is None else {"name": name}` then pass it to `mb` op
2. Have extra if else for `mb` op: `if name is not None: mb.gather(..., name=name); else: mb.gather(...)`

Reply:
Here is the error if we pass `name=None` to `mb` op
```
../_utils.py:71: in _construct_gather_op
    result = mb.gather_nd(x=x, indices=indices, name=name)
../../mil/ops/registry.py:182: in add_op
    return cls._add_op(op_cls_to_add, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'coremltools.converters.mil.mil.builder.Builder'>
op_cls = <class 'coremltools.converters.mil.mil.ops.defs.iOS15.scatter_gather.gather_nd'>
kwargs = {'indices': <coremltools.converters.mil.mil.var.Var object at 0x16c4cc580>, 'x': <coremltools.converters.mil.mil.var.Var object at 0x14e62d690>}
before_op = None

    @classmethod
    def _add_op(cls, op_cls, **kwargs):
        """
        Add an op of type `op_cls` (e.g., convolution) to current block.
        """
        kwargs = cls._maybe_set_name(kwargs, op_cls.__name__)
        logger.debug(
            "Adding op '{}' of type {}".format(kwargs["name"], op_cls.__name__)
        )
        before_op = kwargs.get("before_op", None)
        # Shallow copy list inputs to ensure op inputs are immutable
        kwargs = {k: v if not isinstance(v, (list, tuple)) else v[:] for k, v in kwargs.items() if v is not None}
        kwargs.update(cls._create_vars(
            input_spec=op_cls.input_spec,
>           op_name=kwargs["name"], before_op=before_op,
            candidate_kv=kwargs))
E       KeyError: 'name'

../../mil/builder.py:175: KeyError
```

Reply:
Implemented choice 1